web/src/message_list_data.js

```diff
@@ -590,4 +598,58 @@ export class MessageListData {
         const msg = this._items[msg_index];
         return msg;
     }
+
+    // RECENT VIEW
```

Comment:
We could also do something more like `msg_list_data.recent_view_data.reify_message_id_if_available`, but I chose to try this way because 

(1) `msg_list_data.reify_message_id_if_available` is shorter
(2) I like the idea of having to go through the message list and not being able to access the `recent_view_data` directly. This also lets us access `recent_view_data.conversations` here if we want to, but doesn't let the conversations be accessed more widely.

Reply:
Hmm. `msg_list.data.process_message` is a really evil function name if all it does is update recent view data structures; similarly with `reify_message_id_if_available`; it makes it really sound like it's doing what `echo.update_message_lists` does. I think it's a much nicer abstraction to access `msg_list_data.recent_view_data`... it emphasizes that it's a separate data structure. 

I think we will likely end up with future refactors that make the code paths in `message_events.js` call a single function in `message_list_data` that updates both the recent view and normal data structures -- eliminating the need to have such a `process_message` function called from outside the module at all.

But I think for now, I'd rather we do the overly verbose thing (possibly making use of local variables, like `const recent_view_data = msg_list.recent_view_data`)... and reserve functions exported by this class for things that fully update its data structures (possibly in addition to taking care of recent view stuff).

Reply:
Alright pushed with that change!